pandas/tests/extension/base/ops.py

```diff
@@ -166,23 +166,20 @@ def test_add_series_with_extension_array(self, data):
         expected = pd.Series(data + data)
         tm.assert_series_equal(result, expected)
 
-    @pytest.mark.parametrize("box", [pd.Series, pd.DataFrame])
-    def test_direct_arith_with_ndframe_returns_not_implemented(
-        self, request, data, box
-    ):
-        # EAs should return NotImplemented for ops with Series/DataFrame
+    @pytest.mark.parametrize("box", [pd.Series, pd.DataFrame, pd.Index])
+    def test_direct_arith_with_ndframe_returns_not_implemented(self, data, box):
+        # EAs should return NotImplemented for ops with Series/DataFrame/Index
         # Pandas takes care of unboxing the series and calling the EA's op.
-        other = pd.Series(data)
-        if box is pd.DataFrame:
-            other = other.to_frame()
-        if not hasattr(data, "__add__"):
-            request.node.add_marker(
-                pytest.mark.xfail(
-                    reason=f"{type(data).__name__} does not implement add"
-                )
-            )
-        result = data.__add__(other)
-        assert result is NotImplemented
+        other = box(data)
+
+        op_names = tm.arithmetic_dunder_methods + tm.comparison_dunder_methods
+        op_names = [x for x in op_names if not x.startswith("__r")]
+        for op_name in op_names:
```

Comment:
I prefer the ergonomics parameterizing this so that the test doesn't stop for a failure for just one op

Reply:
ok will update